### ISSUE TITLE
Nested fragment resolution issue

### ISSUE THREAD

**Issue by chanks** [rmosolgo/graphql-ruby#194]
Just upgraded to 0.17.0 and I'm having a problem with fields in fragments sometimes being ignored. Here's a self-contained reproduction (see the comment in the query):

```ruby
require 'graphql'

ElementType = GraphQL::ObjectType.define do
  name("ArrayElement")

  field :value, !types.Int do
    resolve -> (obj, args, ctx) { obj }
  end

  field :times_two, !types.Int do
    resolve -> (obj, args, ctx) { obj * 2 }
  end
end

ArrayType = GraphQL::ObjectType.define do
  name("Array")

  field :first, !ElementType
end

ExampleQueryType = GraphQL::ObjectType.define do
  name("Query")
  field :my_field, ArrayType do
    resolve -> (obj, args, ctx) { [2, 4, 6] }
  end
end

DummySchema = GraphQL::Schema.new(
  query: ExampleQueryType,
)

query_ast = GraphQL.parse <<-GRAPHQL
  query Routes {
    my_field {
      # If we uncomment these lines we lose the times_two field from F0 (!):
      # first {
      #   value
      # }
      ...F3
    }
  }
  fragment F0 on ArrayElement {
    value
    times_two
  }
  fragment F3 on Array {
    first {
      ...F0
    }
  }
GRAPHQL

query = GraphQL::Query.new(
  DummySchema,
  document: query_ast
)

query.result
```

The problem goes away if you move the body of F0 into F3, or F3 into the main query, which makes me think it has something to do with fragments being nested within one another.

**Comment by rmosolgo** [comment]
Thanks for the great writeup! I've got a failing test now -- looks like a bug in rewriting nested fragment spreads which are inside _other_ fragment definitions! 😬

**Comment by rmosolgo** [comment]
Fixed by c9cfa1c, shipped in 0.17.2, could you give it a try?

**Comment by chanks** [comment]
That seems to work! Thanks!
